doc/api18970.md

```diff
@@ -0,0 +1,175 @@
+# Draft material to review in support of [#18970][]
+
+[//]: # (cmark-gfm doc/api18970.md > /tmp/api18970.html)
+
+[#18970]: https://github.com/zephyrproject-rtos/zephyr/issues/18970
+
+## Background
+
+See also the Zephyr [glossary][].
+
+[glossary]: https://docs.zephyrproject.org/latest/glossary.html
+
+### Thread Terminology
+
+Zephyr defines [six thread
+states](https://docs.zephyrproject.org/latest/reference/kernel/threads/index.html#thread-states)
+of which four are active:
+
+* **Ready** when there is nothing that prevents the thread from becoming
+  active as the current thread, but it is not the current thread.
+* **Running** when the thread is active as the current thread (on a
+  processor).
+* **Waiting** when the thread is on a queue waiting for an event to occur
+  that will transition it to *Ready*.
+* **Suspended** when the thread is inactive and must be transitioned to
+  *Ready* through explicitly (via `k_thread_resume()`.
+
+We shall say that a thread **yields** when it invokes a function that
+causes it to become **unready** but not terminated: that is, it
+transitions to *Waiting* or *Suspended*.
+
+**NOTE** The term "sleeping" when applied to Zephyr threads has
+historically mean a thread that is *Waiting* for a timeout event.
+
+Zephyr defines [two classes of
+threads](https://docs.zephyrproject.org/latest/reference/kernel/threads/index.html#thread-priorities):
+
+* **cooperative** threads remain the current thread until they are made
+  unready.
+* **preemptible** threads remain the current thread until either (a) a
+  cooperative thread becomes *Ready*, or (b) a higher-priority
+  preemptible thread becomes *Ready*.
+
+For API behavior we are primarily interested in whether invoking a
+function can cause a cooperative thread to yield.
+
+### Context Terminology
+
+**TODO** fill this out
+
+This section is intended to describe the privilege and processor context
+variations in which a thread can run.  It should provide or reference
+definitions of terms like this:
+
+* Kernel [initialization
+  level](https://docs.zephyrproject.org/latest/reference/drivers/index.html#initialization-levels)
+* User-space versus kernel (system call)
+* Normal ("thread"?) versus interrupt (invoked from an Interrupt Service
+  Routine)
+
+For API behavior we are interested in whether a particular function
+**may**, **must**, or **must not** be invoked from a specific context.
+
+## Definitions
```

Comment:
In particular, that the terms in this section will be added to the glossary

Reply:
Perhaps, perhaps not.  Note that the thread terminology is an an area of documentation about threads; most of this probably belongs in an area of documentation about API specification and behavior.  To be mooted once we have the terminology down.

Reply:
Ditto here, will mark resolved as out of scope for this draft PR, final resolution TBD